jsonnet/thanos-mixin/alerts/querier.libsonnet

```diff
@@ -0,0 +1,58 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'thanos-querier.rules',
+        rules: [
+          {
+            alert: 'ThanosQuerierHighDNSFailures',
+            annotations: {
+              message: 'Thanos Queriers have {{ $value }} of failing DNS queries.',
+            },
+            expr: |||
+              sum(
+                rate(thanos_querier_store_apis_dns_failures_total{%(thanosQuerierSelector)s}[5m])
+              /
+                rate(thanos_querier_store_apis_dns_lookups_total{%(thanosQuerierSelector)s}[5m])
+              ) > 1
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'warning',
+            },
+          },
+          {
+            alert: 'ThanosQuerierInstantLatencyHigh',
+            annotations: {
+              message: 'Thanos Querier has a 99th percentile latency of {{ $value }} seconds for instant queries.',
+            },
+            expr: |||
+              histogram_quantile(0.99,
+                sum(thanos_query_api_instant_query_duration_seconds_bucket{%(thanosQuerierSelector)s}) by (le)
+              ) > 1
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'warning',
+            },
+          },
+          {
+            alert: 'ThanosQuerierRangeLatencyHigh',
+            annotations: {
+              message: 'Thanos Querier has a 99th percentile latency of {{ $value }} seconds for instant queries.',
+            },
+            expr: |||
+              histogram_quantile(0.99,
+                sum(thanos_query_api_range_query_duration_seconds_bucket{%(thanosQuerierSelector)s}) by (le)
+              ) > 1
```

Comment:
I'm fairly certain that we'll hit this often until we introduce the Cortex style caching, but I like the ambition :wink:

Reply:
😂 🚀
What would you suggest? 5s?

Reply:
Let’s keep like this and see how it goes